vertexai/token-count/tokencount_test.go

```diff
@@ -53,3 +53,48 @@ func Test_countTokens(t *testing.T) {
 		t.Errorf("Expected less than 20 tokens, got %d", n)
 	}
 }
+
+func Test_countTokensMultimodal(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	prompt := "Provide a description of the video."
+	video := "gs://cloud-samples-data/generative-ai/video/pixel8.mp4"
+	location := "us-central1"
+	modelName := "gemini-1.5-flash-001"
+
+	var buf bytes.Buffer
+	err := countTokensMultimodal(&buf, prompt, video, tc.ProjectID, location, modelName)
+	if err != nil {
+		t.Fatalf("Test_countTokensMultimodal: %v", err.Error())
+	}
+
+	answer := buf.String()
+
+	for _, expected := range []string{
+		"Number of tokens for the multimodal video prompt: ",
+		"Prompt Token Count:",
+		"Candidates Token Count:",
+		"Total Token Count:",
+	} {
+		if !strings.Contains(answer, expected) {
+			t.Errorf("Response does not contain %q", expected)
+		}
+	}
+
+	s := strings.TrimPrefix(answer, "Number of tokens for the multimodal video prompt: ")
```

Comment:
I wonder why you need to do all this? Why return the answer at all, if this sample is about counting the tokens?

If you need the model's answer, why first join them into a string and then spend so much effort parsing this data back from teh string? The two answers (token count, model response) are conceptually different things anyway

Reply:
At first I was eager to write sample funcs that return proper structured data, but I got pushback because the convention here (on which some automated things rely, iiuc) is to return a single value of type error, and to print any relevant result to the first argument of type `io.Writer`.

This makes assertions more convoluted indeed: either parse a byte stream to extract fields, or give up asserting anything at all on the output.

The sample func and the test func having a different purpose, the priority seems to have a sample func that is equivalent to the Python sample, as concise as practical, and that respects the repo conventions.  For the (less critical) test func, I'm leaning on doing the toilsome effort of asserting the output when possible  (while keeping in mind that models are not fully deterministic).

Reply:
If you're following existing conventions in this repo, that's OK!

Reply:
+1 that this is the convention: https://googlecloudplatform.github.io/samples-style-guide/#result.

I'd probably use a regex to parse the output though.